File: src/tools/users_snapshot/tools.ts
import {McpToolDefContext} from "../common/catoMcpTool.js";
import {buildUsersDetailsTool} from "./usersDetailsTool.js";
import {buildUserConnectionDetailsTool} from "./userConnectionDetailsTool.js";
import {buildClientVersionsTool} from "./clientVersionsTool.js";

export function buildUserSnapshotTools(ctx: McpToolDefContext) {
    return [
        buildUsersDetailsTool(ctx), buildUserConnectionDetailsTool(ctx), buildClientVersionsTool(ctx)
    ];
}

File: python/src/tools/users_snapshot/tools.py
"""
User snapshot tools
"""
from typing import List
from ..common.cato_mcp_tool import CatoMcpToolWrapper, McpToolDefContext
from .users_details_tool import build_users_details_tool
from .user_connection_details_tool import build_user_connection_details_tool
from .client_versions_tool import build_client_versions_tool


def build_user_snapshot_tools(ctx: McpToolDefContext) -> List[CatoMcpToolWrapper]:
    """
    Build all user snapshot tools.
    
    Args:
        ctx: The tool definition context
        
    Returns:
        List of tool wrappers
    """
    return [
        build_users_details_tool(ctx),
        build_user_connection_details_tool(ctx),
        build_client_versions_tool(ctx)
    ]

